kubernetes-model-generator/kubernetes-model-common/src/test/java/io/fabric8/kubernetes/model/jackson/SettableBeanPropertyDelegateTest.java

```diff
@@ -57,45 +57,48 @@ void setUp() {
   @DisplayName("withValueDeserializer, should return a new instance")
   void withValueDeserializer() {
     // Given
-    doReturn(delegateMock).when(delegateMock).withValueDeserializer(any());
+    SettableBeanProperty secondDelegateMock = mock(SettableBeanProperty.class, RETURNS_DEEP_STUBS);
+    doReturn(secondDelegateMock).when(delegateMock).withValueDeserializer(any());
```

Comment:
Jackson is being smart here, and they have a check whether the new delegate is the same as the old, so a separate one needs to be used to test this.